common/src/main/java/io/confluent/connect/storage/common/GenericClassRecommender.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2017 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.connect.storage.common;
+
+import org.apache.kafka.common.config.ConfigDef;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+public class GenericClassRecommender implements ConfigDef.Recommender {
+  private final List<Object> validValues = new ArrayList<>();
+
+  public synchronized void addValidValues(Collection<Object> values) {
```

Comment:
If this is to be a generic _class_ recommender, should the `addValidValues` method accept a collection of `Class` instances? And would be good to genericize this class on the specific type of superclass, e.g., 

```
public class GenericClassRecommender<T>  implements ConfigDef.Recommender {
    ...
    public void addValidValues(Collection<Class<? extends T>> implementationClasses) {
        ...
    }
    ...
}
```

Might also be useful to add a varargs method:

```
    public void addValidValues(Class<? extends T>... implementationClasses) {
        ...
    }
```

Reply:
I didn't bother with generics because the base method `validValues` is not generic anyways.

Reply:
Oh, yeah, I remembered also the real reason. There's an unfortunate layout of classes and interfaces in packages. `Storage` is not visible from common module (it's declare in core). 

I feel that the current change does not justify moving classes around modules at the moment.

Reply:
I think it's really a shame that we can't override ConfigDef configurations, since that would make this so much easier. I'm fine with this, but how about logging an issue to enable redefining configurations?